From 68f489327a18716ff13003e123ec64e538474772 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Thu, 9 Sep 2021 11:36:18 +0200
Subject: [PATCH 0208/1695] core: dt-bindings: add stm32mp tzc400 bindings

Add the TZC400 bindings for STM32MP15 and STM32MP13 memory configuration
in dt-bindings header file. Remove those that were defined in the
platform_config.h header file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 42bfe143f4d55c8ab0f1aa0ea16c64eaded82aaa core: dt-bindings: add stm32mp tzc400 bindings]
Change-Id: I0aa12935da047fbcfbe784a1153024186dc765ff
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/platform_config.h | 13 -------
 core/include/drivers/stm32mp_dt_bindings.h    |  2 ++
 .../dt-bindings/soc/stm32mp-tzc400-macro.h    | 29 +++++++++++++++
 .../dt-bindings/soc/stm32mp13-tzc400.h        | 33 +++++++++++++++++
 .../dt-bindings/soc/stm32mp15-tzc400.h        | 35 +++++++++++++++++++
 5 files changed, 99 insertions(+), 13 deletions(-)
 create mode 100644 core/include/dt-bindings/soc/stm32mp-tzc400-macro.h
 create mode 100644 core/include/dt-bindings/soc/stm32mp13-tzc400.h
 create mode 100644 core/include/dt-bindings/soc/stm32mp15-tzc400.h

diff --git a/core/arch/arm/plat-stm32mp1/platform_config.h b/core/arch/arm/plat-stm32mp1/platform_config.h
index a455dd9b6..c3ff872cd 100644
--- a/core/arch/arm/plat-stm32mp1/platform_config.h
+++ b/core/arch/arm/plat-stm32mp1/platform_config.h
@@ -194,19 +194,6 @@
 #error Inconsistent TAMP backup register zone definition
 #endif
 
-/* TZC resources */
-#define STM32MP1_TZC_A7_ID		U(0)
-#define STM32MP1_TZC_M4_ID		U(1)
-#define STM32MP1_TZC_LTDC_ID		U(3)
-#define STM32MP1_TZC_GPU_ID		U(4)
-#define STM32MP1_TZC_MDMA_ID		U(5)
-#define STM32MP1_TZC_DMA_ID		U(6)
-#define STM32MP1_TZC_USB_HOST_ID	U(7)
-#define STM32MP1_TZC_USB_OTG_ID		U(8)
-#define STM32MP1_TZC_SDMMC_ID		U(9)
-#define STM32MP1_TZC_ETH_ID		U(10)
-#define STM32MP1_TZC_DAP_ID		U(15)
-
 /* USART/UART resources */
 #define USART1_BASE			UART1_BASE
 #define USART2_BASE			UART2_BASE
diff --git a/core/include/drivers/stm32mp_dt_bindings.h b/core/include/drivers/stm32mp_dt_bindings.h
index 62d2a0112..b39dae525 100644
--- a/core/include/drivers/stm32mp_dt_bindings.h
+++ b/core/include/drivers/stm32mp_dt_bindings.h
@@ -10,6 +10,7 @@
 #include <dt-bindings/clock/stm32mp13-clksrc.h>
 #include <dt-bindings/regulator/st,stm32mp13-regulator.h>
 #include <dt-bindings/reset/stm32mp13-resets.h>
+#include <dt-bindings/soc/stm32mp13-tzc400.h>
 #endif
 
 #ifdef CFG_STM32MP15
@@ -17,6 +18,7 @@
 #include <dt-bindings/clock/stm32mp1-clksrc.h>
 #include <dt-bindings/regulator/st,stm32mp15-regulator.h>
 #include <dt-bindings/reset/stm32mp1-resets.h>
+#include <dt-bindings/soc/stm32mp15-tzc400.h>
 #endif
 
 #ifdef CFG_STM32MP25
diff --git a/core/include/dt-bindings/soc/stm32mp-tzc400-macro.h b/core/include/dt-bindings/soc/stm32mp-tzc400-macro.h
new file mode 100644
index 000000000..b697b1e88
--- /dev/null
+++ b/core/include/dt-bindings/soc/stm32mp-tzc400-macro.h
@@ -0,0 +1,29 @@
+/* SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause */
+/*
+ * Copyright (C) 2020-2021, STMicroelectronics - All Rights Reserved
+ */
+
+#ifndef _DT_BINDINGS_STM32MP_TZC400_MACRO_H
+#define _DT_BINDINGS_STM32MP_TZC400_MACRO_H
+
+#define	TZC_REGION_S_NONE	0
+#define	TZC_REGION_S_RD		1
+#define	TZC_REGION_S_WR		2
+#define	TZC_REGION_S_RDWR	3
+
+#define REGION_ID_ACCESS_NSAID_WR_EN_SHIFT	16
+#define REGION_ID_ACCESS_NSAID_RD_EN_SHIFT	0
+#define REGION_ID_ACCESS_NSAID_ID_MASK		0xf
+
+#define TZC_REGION_ACCESS_RD(id)					\
+		((1 << ((id) & REGION_ID_ACCESS_NSAID_ID_MASK)) <<	\
+		 REGION_ID_ACCESS_NSAID_RD_EN_SHIFT)
+
+#define TZC_REGION_ACCESS_WR(id)					\
+		((1 << ((id) & REGION_ID_ACCESS_NSAID_ID_MASK)) <<	\
+		 REGION_ID_ACCESS_NSAID_WR_EN_SHIFT)
+
+#define TZC_REGION_ACCESS_RDWR(id)					\
+		(TZC_REGION_ACCESS_RD(id) | TZC_REGION_ACCESS_WR(id))
+
+#endif /* _DT_BINDINGS_STM32MP_TZC400_MACRO_H */
diff --git a/core/include/dt-bindings/soc/stm32mp13-tzc400.h b/core/include/dt-bindings/soc/stm32mp13-tzc400.h
new file mode 100644
index 000000000..c654716b9
--- /dev/null
+++ b/core/include/dt-bindings/soc/stm32mp13-tzc400.h
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause */
+/*
+ * Copyright (C) 2020-2021, STMicroelectronics - All Rights Reserved
+ */
+
+#ifndef _DT_BINDINGS_STM32MP13_TZC400_H
+#define _DT_BINDINGS_STM32MP13_TZC400_H
+
+/* NSAID */
+#define STM32MP1_TZC_A7_ID		0
+#define STM32MP1_TZC_LCD_ID		3
+#define STM32MP1_TZC_MDMA_ID		5
+#define STM32MP1_TZC_DMA_ID		6
+#define STM32MP1_TZC_USB_HOST_ID	7
+#define STM32MP1_TZC_USB_OTG_ID		8
+#define STM32MP1_TZC_SDMMC_ID		9
+#define STM32MP1_TZC_ETH_ID		10
+#define STM32MP1_TZC_DCMIPP_ID		11
+#define STM32MP1_TZC_DAP_ID		15
+
+#define TZC_REGION_NSEC_ALL_ACCESS_RDWR \
+	(TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_A7_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_LCD_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_MDMA_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_DMA_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_USB_HOST_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_USB_OTG_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_SDMMC_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_ETH_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_DCMIPP_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_DAP_ID))
+
+#endif /* _DT_BINDINGS_STM32MP13_TZC400_H */
diff --git a/core/include/dt-bindings/soc/stm32mp15-tzc400.h b/core/include/dt-bindings/soc/stm32mp15-tzc400.h
new file mode 100644
index 000000000..b704d70a2
--- /dev/null
+++ b/core/include/dt-bindings/soc/stm32mp15-tzc400.h
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause */
+/*
+ * Copyright (C) 2020-2021, STMicroelectronics - All Rights Reserved
+ */
+
+#ifndef _DT_BINDINGS_STM32MP15_TZC400_H
+#define _DT_BINDINGS_STM32MP15_TZC400_H
+
+/* NSAID */
+#define STM32MP1_TZC_A7_ID		0
+#define STM32MP1_TZC_M4_ID		1
+#define STM32MP1_TZC_LCD_ID		3
+#define STM32MP1_TZC_GPU_ID		4
+#define STM32MP1_TZC_MDMA_ID		5
+#define STM32MP1_TZC_DMA_ID		6
+#define STM32MP1_TZC_USB_HOST_ID	7
+#define STM32MP1_TZC_USB_OTG_ID		8
+#define STM32MP1_TZC_SDMMC_ID		9
+#define STM32MP1_TZC_ETH_ID		10
+#define STM32MP1_TZC_DAP_ID		15
+
+#define TZC_REGION_NSEC_ALL_ACCESS_RDWR \
+	(TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_A7_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_GPU_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_LCD_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_MDMA_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_M4_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_DMA_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_USB_HOST_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_USB_OTG_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_SDMMC_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_ETH_ID) | \
+	 TZC_REGION_ACCESS_RDWR(STM32MP1_TZC_DAP_ID))
+
+#endif /* _DT_BINDINGS_STM32MP15_TZC400_H */
-- 
2.47.2

